src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,739 @@
+
+        return $portalTreeNodeDataModel;
+    }
+
+    private function loadChildrenOfNode(string $startNodeId): array
+    {
+        $level = 2; // standard is folder or page
+        if ($startNodeId === $this->rootNodeId) {
+            $level = 1; // children of root are portals
+        }
+
+        $node = new \TdbCmsTree();
+        $node->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+        $node->Load($startNodeId);
+
+        $childrenArray = [];
+        $children = $node->GetChildren(true);
+        while ($child = $children->Next()) {
+            array_push($childrenArray, $this->createTreeDataModel($child, $level));
+        }
+
+        return $childrenArray;
+    }
+
+    private function createTreeDataModel(\TdbCmsTree $node, int $level): BackendTreeNodeDataModel
+    {
+        $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
+
+        $treeNodeDataModel->setName($this->translateNodeName($treeNodeDataModel->getName(), $node));
+        $this->setTypeAndAttributes($treeNodeDataModel, $node);
+
+        // $level 0 == rootNode, 1 = portal, >2 = folder or page
+        if ($level < 2) {
+            $treeNodeDataModel->addListHtmlClass('no-checkbox');
+            $treeNodeDataModel->setOpened(true);
+        }
+
+        ++$level;
+        $children = $node->GetChildren(true);
+        while ($child = $children->Next()) {
+            $childTreeNodeDataModel = $this->createTreeDataModel($child, $level);
+            $treeNodeDataModel->addChildren($childTreeNodeDataModel);
+        }
+
+        return $treeNodeDataModel;
+    }
+
+    private function translateNodeName(string $name, \TdbCmsTree $node): string
+    {
+        $cmsUser = \TCMSUser::GetActiveUser();
+        $editLanguage = $cmsUser->GetCurrentEditLanguageObject();
+        $node->SetLanguage($editLanguage->id);
+
+        if ('' === $name) {
+            $name = $this->global->OutHTML($this->translator->trans('chameleon_system_core.text.unnamed_record'));
+        }
+
+        if (false === CMS_TRANSLATION_FIELD_BASED_EMPTY_TRANSLATION_FALLBACK_TO_BASE_LANGUAGE) {
+            return $name;
+        }
+
+        $cmsConfig = \TCMSConfig::GetInstance();
+        $defaultLanguage = $cmsConfig->GetFieldTranslationBaseLanguage();
+        if (null !== $defaultLanguage && $defaultLanguage->id !== $editLanguage->id) {
+            if ('' === $node->sqlData['name__'.$editLanguage->fieldIso6391]) {
+                $name .= ' <span class="bg-danger px-1"><i class="fas fa-language" title="'.$this->translator->trans('chameleon_system_core.cms_module_table_editor.not_translated').'"></i></span>';
+            }
+        }
+
+        return $name;
+    }
+
+    private function setTypeAndAttributes(BackendTreeNodeDataModel $treeNodeDataModel, \TdbCmsTree $node): void
+    {
+        $treeNodeDataModel->setType($this->setInitialType($node));
```

Comment:
I renamed it to getInitialType()
Initially, the node itself is analyzed, followed by the analysis of whether and which pages are connected. This may change the type again.